### PR TITLE
feat: add support for pre-update and post-update messages

### PR DESCRIPTION
**Pull request by sisp**
I've added support for pre-update and post-update messages that can be written in plain text. Messages can be written inline or included via Jinja includes, and messages are rendered.

This (simple) implementation became possible due to #1156 since there's no output of an internal copy call of the update algorithm anymore. Without this copy call output suppression, the internal copy call would print the pre-copy/post-copy messages in case they are configured which we wouldn't want to see when running an update.

Related to #723. Follow-up of #1194.

### CODE REVIEW

File: tests/test_output.py [hunk 1]
```diff
@@ -131,47 +155,84 @@ def test_message_copy_with_inline_text(
...
+    assert (dst / "version.txt").read_text() == "v2"
+    _, err = capsys.readouterr()
+    assert re.search(
+        (
+            r"^"
+            r"Updating on (linux|macos|windows)"
+            r".+"
+            r"Project {project_name} successfully updated"
+            r"\s*"
+            r"$"
```
Comment (yajo) [inline tests/test_output.py:208]: https://docs.python.org/3/library/re.html#re.VERBOSE

It's beautier to use a verbose regexp IMHO.

File: tests/test_output.py [hunk 2]
```diff
@@ -204,37 +294,72 @@ def test_message_copy_with_included_text(
...
     _, err = capsys.readouterr()
-    assert re.search(pattern.format(project_name="myproj_new"), err, flags=re.S)
+    assert re.search(
+        (
+            r"^"
+            r"Updating on (linux|macos|windows)"
+            r".+"
+            r"Project {project_name} successfully updated"
+            r"\s*"
+            r"$"
```
Comment (yajo) [inline tests/test_output.py:347]: Same here

**Review by yajo — APPROVED** [review]
Looks very nice. Just a suggestion for those regex

File: tests/test_output.py (hunk 1 above)
Comment (yajo) [reply tests/test_output.py:208]: I'd do it slightly different. I think it makes the code clearer TBH:

```py
assert re.search(
    fr"""
    ^Updating\ on\ (linux|macos|windows).+
    Project\ {project_name}\ successfully\ updated\s*$
    """,
    err,
    flags=re.S | re.X,
)
```

Comment (sisp) [reply tests/test_output.py]: Thanks, that looks more concise. :+1: I've added a commit that updates the regexes using verbose syntax also for copy/recopy/update messages. So this PR should probably be rebase-merged, not squash-merged.

**Pull request merged**